lib/config.js

```diff
@@ -9,6 +9,7 @@ config['messages-client'] = config['messages-client'] || {};
 config['messages-client'].clients = config['messages-client'].clients || [];
 config['messages-client'].maxMessageRetrievalRetries =
   config['messages-client'].maxMessageRetrievalRetries || 3;
+config['messages-client'].stopOnMaxRetries = true;
```

Comment:
Maybe we should just ground all of this as "connection failures" because this is a message "client" and that should be sufficient for understanding whether or not you successfully "connect" to get your messages. It's true that there could be some other failues, like some protocol errors, but we can just put those under "connection failures". I'm trying to help think of some names...